src/iter/collect/mod.rs

```diff
@@ -64,6 +66,24 @@ where
     });
 }
 
+/// Collects the iterator into a linked list of vectors.
+///
+/// This is called by `ParallelIterator::collect_vec_list`.
+pub(super) fn collect_vec_list<I>(pi: I) -> LinkedList<Vec<I::Item>>
+where
+    I: ParallelIterator,
+{
+    match pi.opt_len() {
+        Some(len) => {
+            // Pseudo-specialization. See impl of ParallelExtend for Vec for more details.
+            let mut v = Vec::new();
```

Comment:
```suggestion
            let mut v = Vec::with_capacity(len);
```

Reply:
This suggested change is fine, but I think it won't really matter since `special_extend` -> `collect_with_consumer` will call `reserve` anyway. Allocate now or later, but we're still only doing it once.